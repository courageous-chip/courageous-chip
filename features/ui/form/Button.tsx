import { useTheme } from "@react-navigation/native";
import React, { memo } from "react";
import { FlexStyle, StyleSheet, Text, TouchableOpacity } from "react-native";

type Props = { marginBottom?: FlexStyle["marginBottom"]; text: string };

export const Button = memo<Props>(function ({ marginBottom, text }) {
  const {
    colors: { card: backgroundColor, border: borderColor, text: color },
  } = useTheme();

  return (
    <TouchableOpacity
      onPress={() => {}}
      style={[styles.container, { backgroundColor, borderColor, marginBottom }]}
    >
      <Text style={[styles.text, { color }]}>{text}</Text>
    </TouchableOpacity>
  );
});

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    borderWidth: 1,
    justifyContent: "center",
    minHeight: 64,
  },
  text: {
    fontSize: 14,
    lineHeight: 17,
  },
});
