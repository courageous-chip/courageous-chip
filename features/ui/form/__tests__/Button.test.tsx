import React from "react";
import renderer from "react-test-renderer";

import { Button } from "../Button";

describe("<Button />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Button marginBottom={5} text="ðŸŒ´" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders loading", () => {
    const tree = renderer
      .create(<Button loading marginBottom={5} text="ðŸŒ´" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
