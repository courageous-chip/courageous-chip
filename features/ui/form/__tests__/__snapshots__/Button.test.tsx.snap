// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderWidth": 1,
        "justifyContent": "center",
        "minHeight": 64,
      },
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderColor": "rgb(224, 224, 224)",
        "marginBottom": 5,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 14,
          "lineHeight": 17,
        },
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  >
    ðŸŒ´
  </Text>
</TouchableOpacity>
`;
